src/gps/gps/rtcm_pub_node.py

```diff
@@ -0,0 +1,239 @@
+import threading
+import rclpy
+import serial
+from rclpy.node import Node
+from rtcm_msgs.msg import Message as Rtcm
+from pyubx2.ubxtypes_configdb import SET_LAYER_RAM, SET_LAYER_BBR, SET_LAYER_FLASH
+from pyubx2 import (
+    RTCM3_PROTOCOL,
+    UBXMessage,
+    UBXMessageError,
+    UBXParseError,
+    UBXReader,
+    VALCKSUM,
+)
+
+# Defines
+TMODE_SVIN = 1
+TMODE_FIXED = 2
+
+
+class IoManager:
```

Comment:
Separating this class into another module would make code more readable and follow better separation of concerns.

Reply:
The class is specific to UBX protocol specific devices, we could name this something more descriptive like "UBXIOManager"